raiden/tests/unit/transfer/test_node.py

```diff
@@ -56,3 +114,331 @@ def test_is_transaction_effect_satisfied(
+    )
+    assert is_transaction_expired(transaction, block_number)
+
+    transaction = ContractSendSecretReveal(
+        expiration=block_number,
+        secret=factories.UNIT_SECRET,
+        triggered_by_block_hash=factories.make_block_hash(),
+    )
+    assert not is_transaction_expired(transaction, block_number)
+
+
+def test_subdispatch_by_canonical_id(chain_state):
+    our_model, _ = create_model(balance=10, merkletree_width=1)
+    partner_model, _ = create_model(balance=0, merkletree_width=0)
+    channel_state = create_channel_from_models(
+        our_model, partner_model, factories.make_privatekey_bin()
+    )
+    canonical_identifier = channel_state.canonical_identifier
+    token_network = TokenNetworkState(
+        address=canonical_identifier.token_network_address,
+        token_address=factories.make_address(),
+        network_graph=TokenNetworkGraphState(
+            token_network_address=channel_state.token_network_address
+        ),
+    )
+    token_network.partneraddresses_to_channelidentifiers[
+        partner_model.participant_address
+    ] = canonical_identifier.channel_identifier
+    token_network.channelidentifiers_to_channels[
+        canonical_identifier.channel_identifier
+    ] = channel_state
+    payment_network = PaymentNetworkState(
+        address=factories.make_address(), token_network_list=[token_network]
+    )
+    chain_state.identifiers_to_paymentnetworks[payment_network.address] = payment_network
+    chain_state.tokennetworkaddresses_to_paymentnetworkaddresses[
+        canonical_identifier.token_network_address
+    ] = payment_network.address
+    # dispatching a Block will raise an assertion error
+    with pytest.raises(AssertionError):
+        state_change = Block(
+            block_number=chain_state.block_number,
+            gas_limit=GAS_LIMIT,
+            block_hash=chain_state.block_hash,
+        )
+        subdispatch_by_canonical_id(
+            chain_state=chain_state,
+            canonical_identifier=canonical_identifier,
+            state_change=state_change,
+        )
+
+    state_change = ActionChannelClose(canonical_identifier=canonical_identifier)
+
+    # dispatching for an unknown canonical_identifier will not emit events
+    transition_result = subdispatch_by_canonical_id(
+        chain_state=chain_state,
+        canonical_identifier=CanonicalIdentifier(
+            chain_identifier=chain_state.chain_id,
+            token_network_address=factories.make_address(),
+            channel_identifier=factories.make_channel_identifier(),
+        ),
+        state_change=state_change,
+    )
+    assert not transition_result.events, transition_result
+
+    assert get_status(channel_state) == CHANNEL_STATE_OPENED
+    transition_result = subdispatch_by_canonical_id(
+        chain_state=chain_state,
+        canonical_identifier=canonical_identifier,
+        state_change=state_change,
+    )
+
+    assert get_status(channel_state) != CHANNEL_STATE_OPENED
```

Comment:
This check should be exact. In this case, the channel should be closed

Reply:
:+1: `waiting_for_close` to be exact